compiler/rustc_driver_impl/src/lib.rs

```diff
@@ -1237,6 +1238,26 @@ pub fn handle_options(early_dcx: &EarlyDiagCtxt, args: &[String]) -> Option<geto
         return None;
     }
 
+    // To avoid confusion, emit warning if no space
+    // between `-o` and arg, e.g.`-optimize`, `-out-dir`, is applied, see issue #142812
```

Comment:
Suggestion [INTENT 2/2]: actually, probably better to move that doc comment here.